.github/workflows/push-tag-release.yml

```diff
@@ -1,69 +1,263 @@
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+      - name: install poetry
+        shell: bash
+        run: |
+          pip install poetry
+      - name: build style
+        if: ${{ needs.set-variables-create-release.outputs.package == 'qt-style' }}
+        shell: bash
+        run: |
+          pip install -r tools/requirements.txt
+          python tools/build.py build_qt_resources libs/qt-style
+      - name: poetry build
+        shell: bash
+        run: |
+          cd ${{ needs.set-variables-create-release.outputs.folder }}
+          poetry build
+      - name: build standard connect plugin
+        shell: bash
+        # TODO: Remove --from_source when all libraries are published to pypi
+        run: |
+          pip install -r tools/requirements.txt
+          python tools/build.py --remove_intermediate_folder --from_source build_connect_plugin ${{ needs.set-variables-create-release.outputs.folder }}
+      - name: upload artifacts to release
+        shell: bash
+        run: |
+          cd ${{ needs.set-variables-create-release.outputs.folder }}
+          gh release upload ${{ needs.set-variables-create-release.outputs.tag }} dist/*  --clobber
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-  pre-release:
-    needs: pants-build-pre-release
-    uses: ./.github/workflows/release.yml
-    with:
-      changed: ${{ needs.pants-build.outputs.changed }}
-      pre: true
-      version: ${{ needs.pants-build.outputs.version }}
-      deploy_aws: true
-    secrets:
-      aws_key_id: ${{ secrets.S3_ACCESS_KEY }}
-      aws_secret_access_key: ${{ secrets.S3_SECRET_ACCESS_KEY }}
+  publish-test:
+    name: Publish to PyPi (test)
+    runs-on: ubuntu-latest
+    needs: build-linux
+    if: ${{ needs.build-linux.outputs.to_pypi == 'true' }}
+    environment:
+      name: staging
+      url: https://test.pypi.org/project/ftrack-${{ needs.build-linux.outputs.package }}/
+    steps:
+      - name: Download temp artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: dist
+          path: dist
+      - name: Publish to Test PyPi
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          repository_url: https://test.pypi.org/legacy/
 
-  deploy-docs:
-      needs: pants-build
-      uses: ./.github/workflows/deploy-docs.yml
-      with:
-        changed: ${{needs.pants-build.outputs.changed}}
-        build_index: false
-        is_rc: false
\ No newline at end of file
+  publish-prod:
+    name: Publish to PyPi (prod)
+    runs-on: ubuntu-latest
+    needs: build-linux
+    if: ${{ needs.build-linux.outputs.to_pypi == 'true' }}
+    environment:
+      name: production
+      url: https://pypi.org/project/ftrack-${{ needs.build-linux.outputs.package }}/
+    steps:
```

Comment:
I can see in the backend example they have to checkout and setup python, isn't that needed?
Soruce: https://github.com/ftrackhq/ftrack-python/blob/main/.github/workflows/cicd.yml
```
- uses: actions/checkout@v1
      - name: Set up Python
        uses: actions/setup-python@v4
        with:
          python-version: "3.x"
```

Reply:
We don't need this as we already have built the artefact, we just pull it from temp artefact storage